firewood/src/proof.rs

```diff
@@ -102,11 +102,95 @@ const BRANCH_NODE_SIZE: usize = 17;
 /// to a single proof step. If reaches an end step during proof verification,
 /// the hash value will be none, and the encoded value will be the value of the
 /// node.
-pub struct SubProof {
+struct SubProof {
     encoded: Vec<u8>,
     hash: Option<[u8; 32]>,
 }
 
+fn generate_subproof(data: Vec<u8>) -> Result<SubProof, ProofError> {
+    match data.len() {
```

Comment:
nit: `let sub_hash = match data.len() {` then `Ok(SubProof..., Some(sub_hash)` at the end

Reply:
Agree, but same response as above. I think it's easier to see a commit in the history that just moves a function without changing any of the code.